File: Lab3_2/app/src/main/java/com/example/lab3_2/MainActivity.java
package com.example.lab3_2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.method.ScrollingMovementMethod;
import android.view.Gravity;
import android.view.View;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView result = findViewById(R.id.textView3);
        result.setMovementMethod(new ScrollingMovementMethod());

    }

    ArrayList<FuelOrder> fuelOrders = new ArrayList<FuelOrder>();

    public void Sale(View view){
        Spinner fuelType = findViewById(R.id.spinner2);
        EditText value = findViewById(R.id.editTextNumberDecimal2);
        TextView result = findViewById(R.id.textView3);
        if (value.getText().length() == 0) {
            value.setText("0");
        }
        FuelOrder fuelOrder = new FuelOrder(fuelType.getSelectedItem().toString(), Double.parseDouble(value.getText().toString()));
        value.setText("");
        fuelOrders.add(fuelOrder);
        Toast toast = Toast.makeText(getApplicationContext(), "Заказ добавлен", Toast.LENGTH_SHORT);
        toast.setGravity(Gravity.TOP,0,0);
        toast.show();
        //result.setText("Заказ добавлен\n");
    }

    public void Complete(View view){
        TextView result = findViewById(R.id.textView3);
        String res = "";
        for(FuelOrder fuelOrder : fuelOrders){
            res = res + fuelOrder.returnSumOrder();
        }
        result.setText(res);
    }
}

File: Lab2_1/Lab2_1/app/src/main/java/com/example/lab2_1/MainActivity.kt
package com.example.lab2_1
import java.math.*
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import kotlin.math.pow

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val a: EditText = findViewById(R.id.editTextNumber);
        val b: EditText = findViewById(R.id.editTextNumber2);
        val c: EditText = findViewById(R.id.editTextNumber3);
        val res: TextView = findViewById(R.id.textView6)
        val SolveButt: Button = findViewById(R.id.button)
        SolveButt.setOnClickListener{
            val doubleA = a.text.toString().toDouble()
            val doubleB = b.text.toString().toDouble()
            val doubleC = c.text.toString().toDouble()
            res.text = Solver(doubleA,doubleB,doubleC)
        }
    }
}


fun Solver(a:Double,b:Double,c:Double):String {
    val discr: Double = b.pow(2) - 4 * a * c
    when (discr) {
        !in 0.0..Double.MAX_VALUE -> return "Уравнение не имеет рациональных корней"
        0.0 -> return "x = "+(-b)/(2*a)
        in 0.0..Double.MAX_VALUE -> return "x1 = "+(((-b)+discr.pow(0.5))/(2*a)).toString()+",\n x2 = "+(((-b)-discr.pow(0.5))/(2*a)).toString()
    }
    return ""
}